app/models/geo_entity.rb

```diff
@@ -32,19 +34,58 @@ def count_species
     Species.count_species(all_species)
   end
 
-  def occurrences
-    _occurrences = countries_geo_entity_stats.joins(:habitat).select('habitats.name, occurrence')
-    _occurrences = _occurrences.map(&:attributes).map { |attrs| attrs.slice('name', 'occurrence') }
+  def occurrences(real_values = false)
+    # Because of possible nil values for geo entity stat attributes, we can't just go by the occurrence
+    # value; nil values for a particular habitat will translate to data deficient even if confirmed present
+    # as there are two definitions:
+    #
+    # 1) We don't know if the habitat is present at all - this is the 'standard'
+    # data deficient definition
+    # 2) We know the habitat is present, we just don't know how much of it there is
+    occurrences = real_values ? determine_original_occurrences : determine_correct_occurrences
+
     hash = {}
 
-    _occurrences.map do |occurrence|
-      _occurrence = occurrence['occurrence']
+    occurrences.map do |occurrence|
       next if hash[occurrence['name']] == 'present'
-      hash[occurrence['name']] = _occurrence unless _occurrence == 'unknown'
+
+      # unless occurrence['occurrence'] == 'unknown'
+      hash[occurrence['name']] = occurrence['occurrence'] 
     end
+
     GeoEntityStat::BASE_OCCURRENCES.merge(hash)
   end
 
+  def fetch_needed_occurrence_attrs
+    countries_geo_entity_stats.joins(:habitat).select(
+      'habitats.name, 
+      occurrence, 
+      protected_value, 
+      total_value, 
+      geo_entity_stats.protected_percentage'
+    )
+  end
+
+  def determine_original_occurrences
+    occurrences = fetch_needed_occurrence_attrs
+
+    occurrences.map(&:attributes).map { |attrs| attrs.slice('name', 'occurrence') }
```

Comment:
Does this do the same as 

```
occurrences.as_json(only: [:name, :occurrence])
```
?

(as_json is a weird naming... because I think it returns a ruby hash 🤷 )

Reply:
You might get string keys instead of symbols... not sure

Reply:
After looking at my methods, I realise that the `determine_original_occurrences` probably isn't even needed anymore (and it's definitely not being called) with this new approach. I'm just going to remove it altogether. 
As for applying the `as_json` method, you're right that  `map(&:attributes)` and then `slice` is equivalent to it and it is shorter, but it's misleading to read as we're not really outputting a JSON, we're outputting a stringified hash.